src/components/results/results-view-selector/index.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import './style.css';
+
+export const TREE_VIEW = 'tree_view';
+export const JSON_VIEW = 'json_view';
+
+const ResultsViewSelector = ( { view, onViewChange = () => {} } ) =>	{
+	const setTreeView = () => {
+		onViewChange( TREE_VIEW );
+	};
+	const setJsonView = () => {
+		onViewChange( JSON_VIEW );
+	};
+
+	return (
+		<div className={ 'results-view-selector__button-selector-container' }>
+			<button
+				onClick={ setTreeView }
```

Comment:
not sure if this gave you linter warning for creating functions inside of the `onClick` attributes, but given that this is a React function component the same thing the linter might be trying to avoid is still happening, just outside of the attributes.

we could just as easily skip some noise and declare these click handlers inline

```ts
const ResultsViewSelector = ( { … } ) => (
	<div>
		…
		<button onClick={ () => onViewChange( TREE_VIEW ) } … >Tree view</button>
		…
	</div>
);
```